modules/apps/frontend-js/frontend-js-spa-web/npmscripts.config.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+module.exports = {
+	federation: {
+		bridges: ['senna'],
+		runLegacyBuild: false,
+	},
+	preset: '@liferay/npm-scripts/src/presets/standard',
```

Comment:
I'm still confused as to why we need to provide the preset... isn't this the default? If it isn't, why can't we use it? In the past, infra modules had to use a lower-level preset because of the risk of self-consuming imports, but that doesn't feel to be the case anymore, or is it?

Reply:
I will check again, but it was needed because under some circumstances `npm-scripts` didn't load it. I remember to have debugged it, but looking at it now, I don't understand why would it fail removing it, so I'll check again.

Reply:
All `npmscripts.config.js` files in `liferay-portal` define it, otherwise `npm-scripts` fails with:

```
$ liferay-npm-scripts build
Error: getMergedConfig(): property "build" is missing from configuration (existing keys are: ["federation"])
    at pluck (/home/ivan/Liferay/CE/liferay-portal/modules/node_modules/@liferay/npm-scripts/src/utils/getMergedConfig.js:24:9)
    at getMergedConfig (/home/ivan/Liferay/CE/liferay-portal/modules/node_modules/@liferay/npm-scripts/src/utils/getMergedConfig.js:199:9)
    at Object.<anonymous> (/home/ivan/Liferay/CE/liferay-portal/modules/node_modules/@liferay/npm-scripts/src/utils/minify.js:20:22)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
```

I expected it to be inherited too, but since I saw that everybody defines it, I thought it had some purpose... :shrug: 

Summoning @wincent to shed some light on this if possible :pray:

Reply:
BTW, what's the exact semantics of `pluck` :sweat_smile: ?

Reply:
Ah, now I see it. I thought it was an external dep.

Reply:
https://github.com/liferay/liferay-frontend-projects/blob/master/projects/npm-tools/packages/npm-scripts/src/utils/getMergedConfig.js#L160

Reply:
that line says:

> // Use default config if no user config exists

Reply:
It has been like that since the very beginning :point_right: https://github.com/liferay/liferay-frontend-projects/commit/1dd49cb38b146185f3121fe1d45fe79c129593f3

Reply:
Indeed, was going to say it's always been like that.

Reply:
And `pluck` was added [here](https://github.com/liferay/liferay-frontend-projects/commit/7b5f5376d0727d2e38fa7bd2058a3765755a5cbc).

Reply:
Now summoning @bryceosterhaus. Maybe he remembers why overriding npmscripts.config.js disables the preset completely...